pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -159,21 +160,27 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	if checkEngineStateForStop(engine) {
-		reconcileResult, err := r.reconcileForDelete(request)
-		return reconcileResult, err
+	if engine.Instance.Spec.EngineState == "" {
```

Comment:
This is good! but the lower `else if` need to be replaced with `If` otherwise there is no any change from previous and now